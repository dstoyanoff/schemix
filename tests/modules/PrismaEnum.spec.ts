import { describe, expect, it } from "vitest";
import { PrismaEnum } from "schemix/lib/modules/PrismaEnum";

describe("PrismaEnum", () => {
  it("Should support setting the name of a PrismaEnum", () => {
    const foodEnum = new PrismaEnum("Food");
    expect(foodEnum.name).toBe("Food");
  });
  it("Should support adding enum values", async () => {
    const avengersEnum = new PrismaEnum("Avengers")
      .addValue("THOR")
      .addValue("IRON_MAN")
      .addValue("CAPTAIN_AMERICA")
      .addValue("BLACK_WIDOW")
      .addValue("HULK");

    const asString = await avengersEnum.toString();
    expect(asString).toMatchSnapshot();
  });
  it("Should support adding a map to the enum", async () => {
    const testEnum = new PrismaEnum("Food").map("food").addValue("APPLE");

    const asString = await testEnum.toString();
    expect(asString).toMatchSnapshot();
  });
});
