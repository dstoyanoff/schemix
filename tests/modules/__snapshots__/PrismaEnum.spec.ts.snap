// Vitest Snapshot v1

exports[`PrismaEnum > Should support adding a map to the enum 1`] = `
"enum Food {
  APPLE

  @@map(\\"food\\")
}"
`;

exports[`PrismaEnum > Should support adding enum values 1`] = `
"enum Avengers {
  THOR
  IRON_MAN
  CAPTAIN_AMERICA
  BLACK_WIDOW
  HULK
}"
`;
